src/iosrv/srv.c

```diff
@@ -402,6 +402,55 @@ wait_all_exited(struct dss_xstream *dx)
 	D_DEBUG(DB_TRACE, "XS(%d) drained ULTs.\n", dx->dx_xs_id);
 }
 
+/** data structure passed to progress thread when leveraging hyperthread */
+struct dss_progress_info {
+	struct dss_xstream	*dx;
+	struct dss_module_info	*dmi;
+	bool			 should_stop;
+};
+
+/** handler of the progress thread when using SMT */
+static void *
+dss_progress_handler(void *arg)
+{
+	struct dss_progress_info	*info = (struct dss_progress_info *)arg;
+	int				 rc;
+
```

Comment:
need to set TLS (dss_tls_init()).

Reply:
it seems to me that it is not really required since the progress pthread does not use the tls?